Documentation/NvimConfiguration.md

```diff
@@ -0,0 +1,315 @@
+# NVim Project Configuration
+
+NVim can be configured to use the [COC-clangd](https://github.com/clangd/coc-clangd)
+plugin to provide code-completion as well as inline
+[git blame](https://github.com/f-person/git-blame.nvim) using [vim-plug](https://github.com/junegunn/vim-plug).
+
+Make sure you ran `./Meta/serenity.sh rebuild-toolchain` as well as
+`./Meta/serenity.sh run`.
+
+# 0. Install vim-plug
+
+See [https://github.com/junegunn/vim-plug](https://github.com/junegunn/vim-plug).
+
+# 1. Install coc.nvim
+
+The config file for neovim is at `~/.config/nvim/init.vim` or if
+set `$XDG_CONFIG_HOME/nvim/init.vim`.
+
+Add the Plugin:
+```vim
+Plug 'neoclide/coc.nvim', { 'branch': 'release' }
+```
+Run `:PlugInstall` inside nvim.
+
+# 2. Install coc-clangd via CocInstall
+
+```vim
+:CocInstall coc-clangd
+```
+Note: This guide is tested with clangd version 14.0.6
+
+In case you have not installed clangd already install it with
+```
+:CocCommand clangd.install
+```
+
+# 3. Configure coc-clangd in coc-settings.json
+
+Use the following settings to ensure that coc-clangd works out of the box.
+
+Note: You might want to adjust the flalbackFlags or inlayHints.sep according
+to future c++ releases and preference.
```

Comment:
I would split this note up in two parts: One for the hint separator which is purely preference-based, and one for the fallback flags which depend on our current build system and might change in the future.

Reply:
Also, there's a typo in fallbackFlags :^)

Reply:
Done.